src/app/app.context.tsx

```diff
@@ -1,27 +1,28 @@
 import React, { createContext } from "react";
 import { appTypes } from "./app.types";
+import { Theme } from "../enum/theme";
 
 interface AppProviderProps {
   children: React.ReactNode;
 }
 type Dispatch = (action: any) => void;
 
 interface AppProps {
-  theme: string;
+  theme: Theme;
 }
 const initialState: AppProps = {
-  theme: "dark",
+  theme: Theme.DARK,
 };
 const AppStateContext = createContext<AppProps | undefined>(undefined);
 const AppDispatchContext = createContext<Dispatch | undefined>(undefined);
 
 function appReducer(state: AppProps, action: any) {
   switch (action.type) {
     case appTypes.SET_DARK_THEME: {
-      return { ...state, theme: "dark" };
+      return { ...state, theme: Theme.DARK };
     }
     case appTypes.SET_LIGHT_THEME: {
-      return { ...state, theme: "light" };
+      return { ...state, theme: Theme.LIGHT };
     }
     default: {
       throw new Error(`Unhandled action type: ${action.type}`);
```

Comment:
If in case we are using Redux in future we can do as following to avoid long and confusing switch statements. I guess we are not using redux so let it be this way for now.

**Helper**
```javascript

export function createReducer(defaultState: object = {}) {
  const actionHandlerMap = new Map();

  function reducer(state: object = defaultState, action: Action) {
    const actionHandler = actionHandlerMap.get(action.type);

    if (!actionHandler) {
      return state;
    }

    return actionHandler(state, action);
  }

  Object.defineProperty(reducer, 'case', {
    writable: false,
    enumerable: false,
    configurable: false,

    value: function (action: Action) {
      return {
        register: function (handler: Function) {
          actionHandlerMap.set(action, handler);

          return function () {
            actionHandlerMap.delete(action);
          };
        },
      };
    },
  });

  return reducer;
}

export const updateObject = (oldObject: object, updatedProperties: object) => ({
  ...oldObject,
  ...updatedProperties,
});
```
**Use case**
```javascript
export const appReducer = createReducer(initialState);

appReducer.case(ACTION.NAME).register((state,action) => 
     updateObject(state,{
            theme: action.payload
     })
)
```

Reply:
👍  looks cool but we will be avoiding redux